File: searching/Linear Search/Linear_Search.c
/********************************************************************
        Program to search the element linearly in Array
********************************************************************/

#include<stdio.h>
int main()
{
    int Array_1[10],num,key,location;
    int temp = 0;
    
    //Scanning the number of elements of Array_1
    printf("Enter the number of elements of Array_1: ");
    scanf("%d",&num);

    //Scanning the elements of Array_1
    printf("Enter the elements of Array_1: ");
    for(int i=0; i<num; i++)
    {
        scanf("%d",&Array_1[i]);
    }

    //Scanning the element to search in a Array_1
    printf("Enter the element to search in Array_1: ");
    scanf("%d",&key);

    //Checking the key element in the Array_1
    for(int i=0; i<num; i++)
    {
        if(Array_1[i] == key)
        {
            location = i+1;
            temp++;
        }
    }

    //printing the result
    if(temp != 0)
    {
        printf("%d is found at location %d in Array_1",key,location);
    }
    else
    {
        printf("!! %d is not found in Array_1 !!",key);
    }
    return 0;
}

File: Searching/Linear Search/linearSearch.py
# Program to perform linear search

# function to perform linear search on the list
def linearSearch(list_1, size, key):
    for i in range(size):
        if list_1[i] == key:
            return i
    return -1


# scanning the size of the list 
size = int(input("Enter the number of elements to be entered in list: "))

# initialiszing the empty list
list_1 = []

# scanning the elements of the list
print(f"Enter the {size} elements of the list")
for i in range(size):
    print(f"Element {i+1}:", end=" ")
    element = int(input())
    list_1.append(element)

# printing the list
print("Your List:")
print(list_1)

# scanning the key element to find in the list
key = int(input("Enter an element to find in list: "))

# calling the linearSearch() function
result = linearSearch(list_1, size, key)

# printing the conclusion
if result == -1:
    print(f"{key} is not found in the list")
else:
    print(f"{key} is found at position {result + 1} in the list")
